mitocw/edx-platform
# Added REST API for certificates in LMS

#### What are the relevant tickets?
fixes mitocw#230
closes edx#11821 

#### What's this PR do?
Introduces a REST API for the certificates.

#### Where should the reviewer start?
`lms/djangoapps/certificates/apis/v0/views.py`

#### How should this be manually tested?
Create a certificate for a user for a course and, using the url `/api/certificates/v0/certificates/<username>/<course_id>`, check that only that user can see it.
A (fake) certificate can be created using the `GeneratedCertificateFactory`: see 
`lms/djangoapps/certificates/apis/v0/tests/test_views.py` for an example.

#### Any background context you want to provide?
It has been agreed that this API will be deprecated as soon as the [credentials](https://openedx.atlassian.net/wiki/display/AC/Credentials) will be deployed and the existent certificates will be migrated.
For more info please read edx#11821 

#### What gif best describes this PR or how it makes you feel?
![](http://i.giphy.com/uUfbtweIW3A7S.gif)
